Proxy occurs NullPointerException when the SQL is empty String

## Bug Report

### Which version of ShardingSphere did you use?
Sharding-Proxy 4.0.0-RC3-SNAPSHOT

### Expected behavior
When we use IDE, such as Navicat, we can execute the selected SQL by clicking 'Run Selected', but if nothing is selected at the time, the proxy will occur NullPointerException.

As Expected, it should not go wrong

### Reason analyze (If you can)
In SQLParserEngine.parse() method, when the SQL is empty, the ParseTree will be null, and then 
occurs the exception.

The ERROR log is like the following:

> [ERROR]  [ShardingSphere-Command-3] o.a.s.s.f.c.CommandExecutorTask - Exception occur: 
java.lang.NullPointerException: null
	at org.apache.shardingsphere.core.parse.core.parser.SQLParserEngine.parse(SQLParserEngine.java:59)
	at org.apache.shardingsphere.core.parse.core.SQLParseKernel.parse(SQLParseKernel.java:61)
	at org.apache.shardingsphere.shardingproxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:68)

--- a/sharding-proxy/sharding-proxy-backend/src/main/java/org/apache/shardingsphere/shardingproxy/backend/text/TextProtocolBackendHandlerFactory.java
+++ b/sharding-proxy/sharding-proxy-backend/src/main/java/org/apache/shardingsphere/shardingproxy/backend/text/TextProtocolBackendHandlerFactory.java
@@ -17,6 +17,7 @@
 
 package org.apache.shardingsphere.shardingproxy.backend.text;
 
+import com.google.common.base.Strings;
 import lombok.AccessLevel;
 import lombok.NoArgsConstructor;
 import org.apache.shardingsphere.core.parse.core.SQLParseKernel;
@@ -61,6 +62,9 @@ public final class TextProtocolBackendHandlerFactory {
      * @return instance of text protocol backend handler
      */
     public static TextProtocolBackendHandler newInstance(final DatabaseType databaseType, final String sql, final BackendConnection backendConnection) {
+        if (Strings.isNullOrEmpty(sql)) {
+            return new SkipBackendHandler();
+        }
         final String trimSql = SCTLUtils.trimComment(sql);
         if (trimSql.toUpperCase().startsWith(ShardingCTLBackendHandlerFactory.SCTL)) {
             return ShardingCTLBackendHandlerFactory.newInstance(trimSql, backendConnection);

--- a/sharding-proxy/sharding-proxy-backend/src/test/java/org/apache/shardingsphere/shardingproxy/backend/TextProtocolBackendHandlerFactoryTest.java
+++ b/sharding-proxy/sharding-proxy-backend/src/test/java/org/apache/shardingsphere/shardingproxy/backend/TextProtocolBackendHandlerFactoryTest.java
@@ -187,11 +187,4 @@ public final class TextProtocolBackendHandlerFactoryTest {
         TextProtocolBackendHandler actual = TextProtocolBackendHandlerFactory.newInstance(databaseType, sql, backendConnection);
         assertThat(actual, instanceOf(SkipBackendHandler.class));
     }
-    
-    @Test
-    public void assertNewInstanceWithNull() {
-        String sql = null;
-        TextProtocolBackendHandler actual = TextProtocolBackendHandlerFactory.newInstance(databaseType, sql, backendConnection);
-        assertThat(actual, instanceOf(SkipBackendHandler.class));
-    }
 }

--- a/sharding-proxy/sharding-proxy-backend/src/test/java/org/apache/shardingsphere/shardingproxy/backend/TextProtocolBackendHandlerFactoryTest.java
+++ b/sharding-proxy/sharding-proxy-backend/src/test/java/org/apache/shardingsphere/shardingproxy/backend/TextProtocolBackendHandlerFactoryTest.java
@@ -180,4 +180,18 @@ public final class TextProtocolBackendHandlerFactoryTest {
         TextProtocolBackendHandler actual = TextProtocolBackendHandlerFactory.newInstance(databaseType, sql, backendConnection);
         assertThat(actual, instanceOf(QueryBackendHandler.class));
     }
+    
+    @Test
+    public void assertNewInstanceWithEmptyString() {
+        String sql = "";
+        TextProtocolBackendHandler actual = TextProtocolBackendHandlerFactory.newInstance(databaseType, sql, backendConnection);
+        assertThat(actual, instanceOf(SkipBackendHandler.class));
+    }
+    
+    @Test
+    public void assertNewInstanceWithNull() {
+        String sql = null;
+        TextProtocolBackendHandler actual = TextProtocolBackendHandlerFactory.newInstance(databaseType, sql, backendConnection);
+        assertThat(actual, instanceOf(SkipBackendHandler.class));
+    }
 }

